xarray/core/computation.py

```diff
@@ -1006,34 +983,41 @@ def earth_mover_distance(first_samples,
                 f"Please make {(exclude_dims - signature.all_core_dims)} a core dimension"
             )
 
+    # handle dask_gufunc_kwargs
+    if dask == "parallelized":
+        if dask_gufunc_kwargs is None:
+            dask_gufunc_kwargs = {}
+        # todo: remove warnings after deprecation cycle
+        if meta is not None:
+            warnings.warn(
+                "``meta`` should be given in the ``dask_gufunc_kwargs`` parameter."
+                " It will be removed as direct parameter in a future version."
+            )
```

Comment:
Could you please set a class (DeprecationWarning) and stacklevel=2 on these warnings? That results in better messages for users.

Reply:
Sorry to nitpick - shouldn't that be a `FutureWarning` so that users actually get to see it?

Reply:
@mathause At least in the tests the [warnings are issued ](https://dev.azure.com/xarray/xarray/_build/results?buildId=3580&view=logs&j=ba13898e-1dfb-5ace-9966-8b7af3677790&t=2bb5a181-aa33-506a-9515-1f09b3e5c016&l=395). 

What's the actual difference between `DeprecationWarning` and `FutureWarning`? And when should they be used? Just to know for future contributions.

Reply:
FutureWarning would be fine, too. We should probably try to come to consensus on a general policy for xarray.

The Python docs have some guidance but the overall recommendation is not really clear to me: https://docs.python.org/3/library/warnings.html#warning-categories

Reply:
`FutureWarning` is for users and `DeprecationWarning` for library authors (https://docs.python.org/3/library/warnings.html#warning-categories). Which is why you see `DeprecationWarning` in the test but won't when you execute the code. Took me a while to figure this out when I wanted to deprecate some stuff in my package.

```python
import warnings

def test():
    warnings.warn("DeprecationWarning", DeprecationWarning)
    warnings.warn("FutureWarning", FutureWarning)
```

If you try this in ipython `test()` will raise both warnings. But if you save to a file and try

```python
from test_warnings import test
test()
```
only `FutureWarning` will appear (I did not know this detail either https://www.python.org/dev/peps/pep-0565/).

Reply:
@mathause @shoyer I'll switch to `FutureWarning` since this seems to be the only user-visible warning, See https://www.python.org/dev/peps/pep-0565/#id13

Reply:
And, thanks for the pointers and explanations.